src/hooks/useProofsAvailableToMint.ts

```diff
@@ -1,37 +1,24 @@
-import { GoerliContractsStore } from 'stores/ContractsStore'
+import { BadgesContractsStore } from 'stores/ContractsStore'
 import { useSnapshot } from 'valtio'
-import ERC721Proof from 'helpers/ERC721Proof'
-import EmailProof from 'helpers/EmailProof'
 import ProofStore from 'stores/ProofStore'
 import SealCredStore from 'stores/SealCredStore'
+import dataShapeObject from 'helpers/contracts/dataShapeObject'
 import useContractsOwned from 'hooks/useContractsOwned'
 
 export default function () {
-  const contractsOwned = useContractsOwned(GoerliContractsStore)
+  const contractsOwned = useContractsOwned(BadgesContractsStore)
   const { proofsCompleted } = useSnapshot(ProofStore)
-  const { ERC721Ledger, externalERC721Ledger, emailLedger } =
-    useSnapshot(SealCredStore)
+  const { ledgers } = useSnapshot(SealCredStore)
 
-  return proofsCompleted.filter((proof) => {
-    if (proof instanceof EmailProof) {
+  return dataShapeObject((ledgerKey) =>
+    proofsCompleted.filter((proof) => {
+      const proofLedgerName = proof.dataType
+      if (proofLedgerName !== ledgerKey) return false
+      const ledger = ledgers[proofLedgerName]
+      const record = ledger[proof.origin]
       return (
-        !emailLedger[proof.domain] ||
-        !contractsOwned.includes(emailLedger[proof.domain].derivativeContract)
+        !record || !contractsOwned.includes(record.derivative.toLowerCase())
```

Comment:
Can you make a comment here, why we need to do `!`?

Reply:
And maybe we should convert all the `0x...` addresses `toLowerCase` everywhere?

Reply:
I'm not sure what should be convert everywhere, this should be done when we receive/store data from somewhere